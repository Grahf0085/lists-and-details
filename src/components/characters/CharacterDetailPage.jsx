/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import './CharacterDetailPage.css';

const CharDetailPage = ({ name, gender, hair, weapon, profession, position, affiliation }) => (

  <div>
    <p>Name: {name}</p>
    <p>Gender: {gender}</p>
    <p>Hair: {hair}</p>
    <p>Weapon: {weapon}</p>
    <p>Job: {profession}</p>
    <p>Position: {position}</p>
    <p>Party: {affiliation}</p>
  </div>

);

CharDetailPage.propTypes = {
  name: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  hair: PropTypes.string.isRequired,
  weapon: PropTypes.string.isRequired,
  profession: PropTypes.string.isRequired,
  position: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired
};

export default CharDetailPage;
