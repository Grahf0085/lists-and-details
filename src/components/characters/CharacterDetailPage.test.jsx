import React from 'react';
import { render } from '@testing-library/react';
import CharacterDetailPage from './CharacterDetailPage';

describe('Character component', () => {
  it('renders details', () => {
    const { asFragment } = render(
      <CharacterDetailPage
        name="Text Character"
        gender="983247fdssf"
        hair="http://image.com/image.png"
        weapon="Text Character"
        profession="983247fdssf"
        position="http://image.com/image.png"
        affiliation="Text Character"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
