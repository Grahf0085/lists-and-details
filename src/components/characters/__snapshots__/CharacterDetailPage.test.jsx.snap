// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders details 1`] = `
<DocumentFragment>
  <div
    role="omg"
  >
    <p>
      Name: Text Character
    </p>
    <p>
      Gender: 983247fdssf
    </p>
    <p>
      Hair: http://image.com/image.png
    </p>
    <p>
      Weapon: Text Character
    </p>
    <p>
      Job: 983247fdssf
    </p>
    <p>
      Position: http://image.com/image.png
    </p>
    <p>
      Party: Text Character
    </p>
  </div>
</DocumentFragment>
`;
